pkg/image/apis/image/v1/types.go

```diff
@@ -184,7 +184,9 @@ type ImageStreamSpec struct {
 	// Specify the source for the tags to be imported in each tag via the spec.tags.from reference instead.
 	DockerImageRepository string `json:"dockerImageRepository,omitempty" protobuf:"bytes,1,opt,name=dockerImageRepository"`
 	// tags map arbitrary string values to specific image locators
-	Tags []TagReference `json:"tags,omitempty" protobuf:"bytes,2,rep,name=tags"`
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	Tags []TagReference `json:"tags,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,2,rep,name=tags"`
```

Comment:
Can we add a test to test-cmd that verifies this merges correctly?

Reply:
done.

Reply:
> Can we add a test to test-cmd that verifies this merges correctly?

Turns out, that was a really good idea.

Reply:
no need to encourage him...